extended/src/main/java/io/kubernetes/client/extended/leaderelection/LeaderElector.java

```diff
@@ -334,8 +343,17 @@ private void maybeReportTransition() {
     }
   }
 
-  private ThreadFactory makeThreadFactory(String nameFormat) {
-    return new ThreadFactoryBuilder().setDaemon(true).setNameFormat(nameFormat).build();
+  private ThreadFactory makeThreadFactory(String format) {
+    final ThreadFactory defaultFactory = Executors.defaultThreadFactory();
+    final AtomicInteger threadNumber = new AtomicInteger(1);
```

Comment:
This code is repeated, please factor out into a single method that is re-used in various places.

Reply:
It's not repeated in this class. Do you mean you want to have a utility class that is shared across the Java SDK?

Reply:
Did that.